statsmodels/discrete/discrete_model.py

```diff
@@ -1882,8 +2026,36 @@ def score_obs(self, params):
 
         y = self.endog
         X = self.exog
-        L = self.cdf(np.dot(X, params))
-        return (y - L)[:,None] * X
+        fitted = self.predict(params)
+        return (y - fitted)[:,None] * X
+
+    def score_factor(self, params):
+        """
+        Logit model derivative of the log-likelihood with respect to linpred
```

Comment:
same comment as with hessian_factor above

Reply:
similar answer
score_factor is 1-D and can be multiplied by any "instruments" or other exog not in the model.